File: abc394/b/main.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main(){
    int n;
    cin >> n;
    vector<string> s(n);
    vector<pair<int,int>> id(n);
    for(int i = 0; i < n; i++){
        cin >> s[i];
        id[i].first = s[i].size();
        id[i].second = i;
    }
    sort(id.begin(), id.end());
    for(int i = 0; i < n; i++) cout << s[id[i].second] << (i == n-1 ? "\n" : "");
}

File: abc394/b/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
)

type Pair struct {
	first  any
	second any
}

func main() {
	r := bufio.NewReader(os.Stdin)
	w := bufio.NewWriter(os.Stdout)
	defer w.Flush()
	var n int
	fmt.Fscan(r, &n)
	s := make([]Pair, n)
	for i := 0; i < n; i++ {
		var str string
		fmt.Fscan(r, &str)
		s[i].first = len(str)
		s[i].second = str
	}
	sort.Slice(s, func(i, j int) bool {
		return s[i].first.(int) < s[j].first.(int)
	})
	for _, v := range s {
		fmt.Fprintf(w, "%s", v.second)
	}
	fmt.Fprintln(w)
}
